sdk/dotnet/Pulumi/Config_Options.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright 2016-2018, Pulumi Corporation
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Pulumi
+{
+    public partial class Config
+    {
+        /// <summary>
+        /// StringConfigOptions may be used to constrain the set of legal values a string config value may contain.
+        /// </summary>
+        public class StringOptions

```

Comment:
Do we want to nest these inside `Config`? Doesn't that mean usage is going to be:

```c#
var foo = config.Get("foo", new Config.StringOptions { MaxLength = 10 });
```

Rather than shorter:

```c#
var foo = config.Get("foo", new StringOptions { MaxLength = 10 });
```

Reply:
I basically don't want these cluttering the top level pulumi namespace.  they could be in an inner namespace, or nested like this.  Frankly, i think they're fairly pointless.  But i'm keeping parity with our JS sdk here.  Having someone just validate their own string values themselves is pretty trivial (and can be added with extension methods tbh).

Reply:
removed.